### ISSUE TITLE
`#error?(path)` doesn't seem to work correctly

### ISSUE THREAD

**Issue by pigats** [dry-rb/dry-schema#317]
**Describe the bug**

`#error?(path)` doesn't seem to work correctly with nested validations

**To Reproduce**

```ruby
require "dry-schema"
require "json"
require "rspec"

class Schema
  BlockItem = Dry::Schema.JSON do
    optional(:payload).filled(:hash)
  end

  Item = Dry::Schema.JSON do
    required(:i18n).array(:hash) do
      required(:blocks).array(BlockItem)
    end
  end

  Items = Dry::Schema.JSON do
    required(:items).array(Item)
  end
end

payload = JSON.parse(
<<~JSON
  {
    "items": [
      {
        "i18n": [
          {
            "blocks": [
              {
                "payload": {
                  "key": "value"
                }
              },
              {
                "payload": {}
              }
            ]
          }
        ]
      },
      {
        "i18n": [
          {
            "blocks": [
              {
                "payload": {
                  "key": "value"
                }
              }
            ]
          }
        ]
      }
    ]
  }
JSON
)

RSpec.describe("#error?(path)") do
  it "is true, false" do
    result = Schema::Items.(payload)
    expect(result.error?([:items, 0])).to be true
    expect(result.error?([:items, 1])).to be false
  end
end
```

**Expected behavior**

`result.error?([:items, 1])` should be `false`, as the second item is valid.

**Notes**

Please note that changing the position of the invalid block affects the result - it seems the index passed to `#error?([:items, index])` is checking the block position rather than the item position - eg. with
 
```json
 {
    "items": [
      {
        "i18n": [
          {
            "blocks": [
              {
                "payload": {
                  "key": "value"
                }
              },              
              {
                "payload": {
                  "key": "value"
                }
              },
              {
                "payload": {}
              }
            ]
          }
        ]
      },
      {
        "i18n": [
          {
            "blocks": [
              {
                "payload": {
                  "key": "value"
                }
              }
            ]
          }
        ]
      }
    ]
  }
```

`result.error?([:items, 1])` is `false` and `result.error?([:items, 2])` is `true`, although there’s no 3rd item in the `items` array.

**Your environment**

- Affects my production application: Yes
- Ruby version:  2.6.5p114
- OS: CentOS Linux release 7.4.1708
- dry-schema: 1.5.4

**Comment by solnic** [comment]
@pigats thanks for the detailed report Andrea <3 I'll fix it later today and push a release.

**Comment by solnic** [comment]
oops, I fixed this case but broke the one that worked before 🙄

**Comment by solnic** [comment]
@pigats OK I just released 1.5.5 with the fix - pls lemme know if it works for you

**Comment by pigats** [comment]
thanks @solnic, it works 🙌
